app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/Filter.kt

```diff
@@ -263,3 +323,142 @@ fun FilterSection(
+    textFieldValue: MutableState<String>
+) {
+
+  // Initialize FocusManager
+  val focusManager = LocalFocusManager.current
+
+  // Initialize the keyboard controller
+  val virtualKeyboardManager = LocalSoftwareKeyboardController.current
+
+  // List of suggestions from NominatimAPI
+  val listOfSuggestions = addressViewModel.addressList.collectAsState()
+
+  val uniqueSuggestions = remember { mutableStateOf(listOf<Address>()) }
+
+  // Show suggestions screen
+  val showSuggestions = remember { mutableStateOf(false) }
+
+  // value that contain the size fo the textfield to align the suggestion menu below
+  val textFieldSize = remember { mutableStateOf(IntSize.Zero) }
+
+  val alpha by animateFloatAsState(targetValue = if (isTextFieldVisible.value) 1f else 0f)
+
+  // Animation values
+  val slideOffset by
+      animateDpAsState(targetValue = if (isTextFieldVisible.value) 0.dp else (-200).dp)
+
+  Box {
+    OutlinedTextField(
+        value = textFieldValue.value,
+        onValueChange = { textFieldValue.value = it },
+        placeholder = { Text(text = stringResource(R.string.search_bar_placeholder)) },
+        modifier =
+            Modifier.padding(start = 8.dp)
+                .offset(x = slideOffset)
+                .alpha(alpha)
+                .onGloballyPositioned { coordinates -> textFieldSize.value = coordinates.size }
+                .testTag("SearchBarListScreen"),
+        colors = getOutlinedTextFieldColorsSearchBar(ColorLevel.PRIMARY),
+        trailingIcon = {
+          if (textFieldValue.value.isNotEmpty()) {
+            Icon(
+                imageVector = Icons.Filled.Clear,
+                contentDescription = "Clear search",
+                tint = defaultOnColor(),
+                modifier =
+                    Modifier.clickable {
+                          textFieldValue.value = ""
+                          showSuggestions.value = false
+                        }
+                        .testTag("ClearSearchButtonListScreen"))
+          }
+        },
+        singleLine = true,
+        keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Search),
+        keyboardActions =
+            KeyboardActions(
+                onSearch = {
+                  virtualKeyboardManager?.hide()
+
+                  runBlocking { addressViewModel.search(textFieldValue.value) }
+
+                  showSuggestions.value = true
+                }))
+
+    if (listOfSuggestions.value.size + 1 > 0) {
+      DropdownMenu(
+          expanded = showSuggestions.value,
+          onDismissRequest = { showSuggestions.value = false },
+          modifier =
+              Modifier.width(with(LocalDensity.current) { textFieldSize.value.width.toDp() })
+                  .testTag("SuggestionsMenuListScreen")) {
+            if (permissionHandler.getLocalisationPerm().collectAsState().value) {
+              DropdownMenuItem(
+                  text = { Text("My Location") },
```

Comment:
Here you forgot to use a string resource, which you could reuse on line 422

Reply:
done !